### DISCUSSION TITLE
large objects (1.2 gigabytes) did not get mirrored to the destination amazon s3 from self-hosted minio s3

### DISCUSSION THREAD

**Discussion by Sina-Najafi** [minio/minio#18524] [Q&A]
Hi,
I tried to mirror the contents of a bucket that I have in my self-hosted minio s3 to an amazon s3 bucket. I use the bucket as a DVC repository. to do so, I used the command `mc mirror --watch <source> <destination>`. I got similar error messages for a few objects. I will copy paste a couple of them:

mc: <ERROR> Failed to copy `https://s3.mydomain.com/dvc-bucket/files/md5/de/dca4f59ef6217705f42eb1eb60cba2`. Put "https://dvc-bucket-sherkat.s3.dualstack.us-east-1.amazonaws.com/files/md5/de/dca4f59ef6218805f42eb1eb60cba2?partNumber=67&uploadId=alo2h_klbUeGeuviQigiIfiRL.j3lC9hwJBEO4Dm0qYigZMEAZ8BNDwTAVkFlPx3vZ3RKdVAvpMbTQ0o1XTNVYkDfAJe1AfgA49U9doW.4qyCZjupR_vZYwPbPpweua6": net/http: HTTP/1.x transport connection broken: http: ContentLength=16777216 with Body length 224344

mc: <ERROR> Failed to copy `https://s3.mydomain.com/dvc-bucket/files/md5/62/145573141f426e8a63f603f84a1183`. Put "https://dvc-bucket-sherkat.s3.dualstack.us-east-1.amazonaws.com/files/md5/62/145589141f426e8a63f603f84a1183?partNumber=67&uploadId=d2PpUo6c0lRfs3HOAroR8EAE8xAQ348T86N05mRkacNNv75bOs9f1dKWgi43g1Qw1xXAOJqs8vlls9JwvCaiJKLCr9DK2jgXjPoY0srWuIlCikcoMCS6DW1nBapHNC4s": net/http: HTTP/1.x transport connection broken: http: ContentLength=16777216 with Body length 2729616

in my desitination amazon s3 bucket I can't find the objects that were in the errors.
I noticed in my source bucket that these objects that failed to copy to the distination bucket are large compared to other objects(they are in order of 1 or 2 gigabytes).
how do I fix this? the user I use for accessing the amazon s3 bucket has full s3 access. and I don't think I changed any default setting. is there any way I can fix this and mirror those files too?? can any default setting prevent large files from copying? do you think using aws cli instead of mc will fix my issue? I need help!
thanks
I don't know if it helps but ContentLength=16777216 is always the same in the errors. so is the part number

**Comment by klauspost** [comment]
Since it is "Put" operations that are failing the issue would seem to be on AWS side.
